vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/treegrid/TreeGrid.java

```diff
@@ -326,20 +214,54 @@ public TreeGrid(HierarchicalDataProvider<T, ?> dataProvider) {
         setDataProvider(dataProvider);
     }
 
+    private static class TreeGridDataCommunicator<T>
+            extends HierarchicalDataCommunicator<T> {
+        private Element element;
+
+        public TreeGridDataCommunicator(Element element,
+                CompositeDataGenerator<T> dataGenerator,
+                ArrayUpdater arrayUpdater,
+                SerializableConsumer<JsonArray> dataUpdater,
+                SerializableSupplier<ValueProvider<T, String>> uniqueKeyProviderSupplier) {
+            super(dataGenerator, arrayUpdater, dataUpdater, element.getNode(),
+                    uniqueKeyProviderSupplier);
+            this.element = element;
+        }
+
+        @Override
+        public void reset() {
+            super.reset();
+            if (element != null) {
+                element.callJsFunction("$connector.reset");
```

Comment:
This replaces the `$connector.ensureHierarchy` call which was previously made from the data communicator class itself.